crates/rspack_core/src/build_chunk_graph/code_splitter.rs

```diff
@@ -322,24 +324,30 @@ impl<'me> CodeSplitter<'me> {
 
   fn process_module(&mut self, item: &QueueItem) {
     tracing::trace!("process_module {:?}", item);
+
     let mgm = self
       .compilation
       .module_graph
       .module_graph_module_by_identifier(&item.module_identifier)
       .unwrap_or_else(|| panic!("no module found: {:?}", &item.module_identifier));
 
-    for module_identifier in mgm
-      .depended_modules(&self.compilation.module_graph)
-      .into_iter()
-      .rev()
-    {
-      self.queue.push(QueueItem {
-        action: QueueAction::AddAndEnter,
-        chunk: item.chunk,
-        chunk_group: item.chunk_group,
-        module_identifier: *module_identifier,
+    let queue_items = self
+      .depended_modules_cache
+      .entry(item.module_identifier)
+      .or_insert_with(|| {
+        mgm
+          .depended_modules(&self.compilation.module_graph)
+          .into_iter()
+          .rev()
+          .map(|module_identifier| QueueItem {
+            action: QueueAction::AddAndEnter,
+            chunk: item.chunk,
+            chunk_group: item.chunk_group,
```

Comment:
Good works!

https://github.com/web-infra-dev/rspack/blob/70519c86eda8a802af7e0602afb3777ffad3c33c/crates/rspack_core/src/build_chunk_graph/code_splitter.rs#L344-L345

But the captured values in the above lines are incorrect. It would add `Module` to the unexpected `Chunk` while hitting the cache.

I suggest you could only cache the modules, such like 
 `depended_modules_cache: IdentifierMap<Vec<ModuleIdentifer>>`

Reply:
@hyf0 Done. The performance gain is still significant.